crates/recursor/src/recursor_dns_handle.rs

```diff
@@ -333,15 +335,18 @@ impl<P: ConnectionProvider> RecursorDnsHandle<P> {
         Ok(response)
     }
 
-    async fn lookup(
+    /// Retrieve a response from the cache, optionally ignoring responses that lack DNSSEC records.
+    fn filtered_cache_lookup(
         &self,
-        query: Query,
-        ns: RecursorPool<P>,
+        query: &Query,
         now: Instant,
         expect_dnssec_in_cached_response: bool,
-    ) -> Result<Message, Error> {
-        if let Some(response_res) = self.response_cache.get(&query, now) {
-            let response = response_res?;
+    ) -> Option<Result<Message, Error>> {
+        if let Some(response_res) = self.response_cache.get(query, now) {
```

Comment:
I think we can now invert the condition to return early for `None`?

(I think this happens in the last commit, but changes would be easier to review if you did it here on the first introduction of `filtered_cache_lookup()` IMO.)